src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs

```diff
@@ -159,36 +234,53 @@ private async Task<bool> UseGoCliToScanAsync(string location, ISingleFileCompone
         return true;
     }
 
-    private void ParseGoModFile(
+    private async Task ParseGoModFileAsync(
         ISingleFileComponentRecorder singleFileComponentRecorder,
         IComponentStream file,
         GoGraphTelemetryRecord goGraphTelemetryRecord)
     {
-        using var reader = new StreamReader(file.Stream);
-
-        var line = reader.ReadLine();
-        while (line != null && !line.StartsWith("require ("))
+        void TryRegisterDependency(string s)
```

Comment:
nit: pick a better name for this parameter, also consider moving the method outside. There are some advantages of using nested methods like sharing context but this method seems pretty independent as it is. I don't see a reason for having a nested declaration in this case.